### PR TITLE
Refactor and create `data` module

### PR DESCRIPTION
**Pull request by dhpitt**
Old structure:

```
neuralop
| --- datasets
        | --- transforms
        | --- data_transforms
        | --- darcy
        | --- etc
```

New structure:

```
neuralop
| --- data
        | --- datasets
                | --- darcy
                | --- etc
        | --- transforms
                | --- normalizers
                | --- etc
```

### CODE REVIEW

File: neuralop/data/transforms/base_transforms.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    @abstractmethod
+    def cpu(self):
+        pass
+
+    @abstractmethod
+    def to(self, device):
+        pass
+
+class Composite(Transform):
```
Comment (JeanKossaifi) [inline neuralop/data/transforms/base_transforms.py:34]: Should this just be named Compose?

File: neuralop/data/transforms/normalizers.py [hunk 2]
```diff
@@ -242,3 +193,57 @@ def from_dataset(cls, dataset, dim=None, keys=None, mask=None):
                 if key in keys:
                     instances[key].partial_fit(sample.unsqueeze(0))
         return instances
+
+class MultipleFieldUnitGaussianNormalizer(DictTransform):
```
Comment (JeanKossaifi) [inline neuralop/data/transforms/normalizers.py:197]: Should we name this just DictGaussianNormalizer?

File: neuralop/data/transforms/positional_embeddings.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
+from typing import List
+
+import torch
+from torch import nn
+
+from .grid_transforms import regular_grid_2d, regular_grid_nd
+
+class Euclidean2D(nn.Module):
```
Comment (JeanKossaifi) [inline neuralop/data/transforms/positional_embeddings.py:8]: I'd name this just Regular2DGrid, that would be clearer to users.

Comment (JeanKossaifi) [reply neuralop/data/transforms/positional_embeddings.py:8]: Or maybe Embedding rather than Grid so it's consistent for sinusoidal case

File: neuralop/data/transforms/transforms.py [hunk 4]
```diff
@@ -73,6 +64,79 @@ def to(self, device):
         self.transforms = [t.to(device) for t in self.transforms if hasattr(t, 'to')]
         return self
 
+class DictTransform(Transform):
```
Comment (JeanKossaifi) [inline neuralop/data/transforms/transforms.py:67]: Why is this duplicated? We already have it in base_transforms no?

**Review by JeanKossaifi — COMMENTED** [review]
Thanks for the changes @dhpitt! Left some comments.

File: neuralop/data/transforms/positional_embeddings.py (hunk 3 above)
Comment (dhpitt) [reply neuralop/data/transforms/positional_embeddings.py:8]: What about Regular2DGridEmbedding? Too wordy?

File: neuralop/data/transforms/transforms.py (hunk 4 above)
Comment (dhpitt) [reply neuralop/data/transforms/transforms.py]: Yeah it was, just removed this file

File: neuralop/data/transforms/positional_embeddings.py (hunk 3 above)
Comment (JeanKossaifi) [reply neuralop/data/transforms/positional_embeddings.py:8]: If its Regular it's probably a grid? How about just GridEmbedding? (or GridEmbedding2D)

Comment (dhpitt) [reply neuralop/data/transforms/positional_embeddings.py]: That sounds good. Updated.

**Comment by JeanKossaifi** [comment]
Looks great, thank you @dhpitt! 
Merging.

**Comment by matteoguarrera** [comment]
docs still need to be fixed :smile:  I opened an issue #341 i can do it if you want guys
